napari/components/viewer_model.py

```diff
@@ -402,9 +402,9 @@ def _new_labels(self):
         extent = layers_extent.world
         scale = layers_extent.step
         scene_size = extent[1] - extent[0]
-        corner = extent[0] + 0.5 * layers_extent.step
+        corner = extent[0]
         shape = [
-            np.round(s / sc).astype('int') if s > 0 else 1
+            (np.round(s / sc).astype('int') if s > 0 else 1) + 1
```

Comment:
Can scene_size be negative? If not, I think you simplify to the following.

```suggestion
            (np.round(s / sc).astype('int')) + 1
```

Otherwise, maybe it should be `else 0` now? Or `max(np.round(s / sc).astype('int'), 0)`?

Reply:
I think you're right that it can't be negative.

Reply:
I'll try without first.